kedro/io/data_catalog.py

```diff
@@ -293,6 +293,15 @@ class to be loaded is specified with the key ``type`` and their
                 dataset_patterns[ds_name] = ds_config
 
             else:
+                # Check if 'layer' attribute is defined at the top level
+                if "layer" in ds_config:
+                    import warnings
+
+                    warnings.warn(
+                        "Defining the 'layer' attribute at the top level is deprecated "
+                        "and will be removed in Kedro 0.19.0. Please move 'layer' inside the 'metadata' attribute.",
```

Comment:
Actually, it needs to be moved under `kedro-viz` under `metadata` so just link it to the new way if we have it in the docs